boot.php

```diff
@@ -785,60 +785,85 @@ function get_scheme() {
 		return($this->scheme);
 	}
 
+	/**
+	 * @brief Retrieves the Friendica instance base URL
+	 *
+	 * Caches both SSL and non-SSL version for performance
+	 *
+	 * @param bool $ssl
+	 * @return string
+	 */
 	function get_baseurl($ssl = false) {
 
 		// Is the function called statically?
-		if (!is_object($this))
-			return(self::$a->get_baseurl($ssl));
+		if (!is_object($this)) {
+			return self::$a->get_baseurl($ssl);
+		}
 
-		$scheme = $this->scheme;
+		if (!isset($this->baseurl[$ssl ? 'https' : 'http'])) {
```

Comment:
I don't mind adding a comment even if it feels redundant to me, I was just wondering what you meant about the cache key and if I needed to change the actual code.